### PR TITLE
Add KeyId type

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Add KeyId type which provides an idiomatic rust wrapper for the
`sgx_key_id_t` type.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #80** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/80" target="_blank">Graphite</a> 
    * **PR #81** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/81" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/81?utm_source=stack-comment).

File: core/sys/types/src/lib.rs [hunk 1]
```diff
@@ -11,3 +11,9 @@
 )]
 
 include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+impl Clone for sgx_key_id_t {
+    fn clone(&self) -> sgx_key_id_t {
+        sgx_key_id_t { id: self.id }
+    }
+}
```
Comment (jcape) [inline core/sys/types/src/lib.rs:19]: This should be done by an `mc_sgx_core_build::SgxParseCallbacks::add_derives()` implementation...

**Review by jcape — APPROVED** [review]
Looks good generally, but I suggest using `ParseCallbacks::add_derives()` to make everything `#[derive(Clone)]`

File: core/sys/types/src/lib.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply core/sys/types/src/lib.rs]: @jcape made an attempt in https://github.com/mobilecoinfoundation/sgx/pull/81/commits/bb63bbdd2766b2b5302bf66824b6fa1111f2ad37 let me know thoughts.
I don't like putting *all* the types in the `build/lib.rs`, perhaps `sgx_builder()` should be updated to take a `non-cloneable` iterator.  I don't like negative args, but it seems to be the narrower scope.

Comment (jcape) [reply core/sys/types/src/lib.rs]: We should definitely set `derive_copy(false)` in the `sgx_builder()`, that was me simply not thinking.

What we can do is make `SgxParseCallbacks { do_not_clone: Vec<String> }` and `SgxParseCallbacks::new(do_not_clone: impl Iter<Item=&str>)`, and put that into the `Box`.

Comment (nick-mobilecoin) [reply core/sys/types/src/lib.rs]: next attempt at, https://github.com/mobilecoinfoundation/sgx/pull/81/commits/a824269f393775f66fd3d7760bdbcd5e37b84a19

**Review by awygle — APPROVED** [review]
I approve of this as an implementation, if we decide we want a KeyId wrapper.

**Comment by nick-mobilecoin** [comment]
Closing as KeyId doesn't seem like it's needed as a rust type at this time

**Pull request closed without merging**